specutils/spectra/spectral_coordinate.py

```diff
@@ -521,18 +527,17 @@ def _change_observer_to(self, observer, target=None):
         init_obs_vel = self._calculate_radial_velocity(self.observer, target)
         fin_obs_vel = self._calculate_radial_velocity(observer, target)
 
-        # TODO: getting proper blueshifted results in the test involving
-        #  shifting an observer to the target velocity frame requires that we
-        #  negate the initial velocity between observer/target. Unsure why.
-        new_data = self._project_velocity_and_shift(-init_obs_vel, fin_obs_vel)
+        line_of_sight_unit_vec = self._norm_d_pos(observer, target).xyz
+
+        new_data = self._project_velocity_and_shift(init_obs_vel, fin_obs_vel, line_of_sight_unit_vec)
 
         new_coord = self._copy(value=new_data,
                                observer=observer,
                                target=target)
 
         return new_coord
 
-    def _project_velocity_and_shift(self, init_vel, fin_vel):
+    def _project_velocity_and_shift(self, init_vel, fin_vel, line_of_sight_unit_vec):
```

Comment:
The issue with this method was that it actually had no concept of the direction between observer and target since both or either velocity vectors could be in the opposite direction. So we need to pass in a separate vector giving the actual observer to target direction with the correct orientation.